### PR TITLE
Remove java Dockerfile to build

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Part of #740, now we can rely on local setup for prow

## Proposed Changes

- :broom: Removed Java Dockerfile to build

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
Seems like java is still not installed...

**Comment by slinkydeveloper** [comment]
It works!

**Comment by slinkydeveloper** [comment]
Java unit tests are running, but golang unit test are not :thinking: https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/829/pull-knative-sandbox-eventing-kafka-broker-unit-tests/1385877919519215616#1:build-log.txt%3A140

Also, the tests report doesn't show the log lines related to each test (not a big problem)

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -35,14 +35,11 @@ function fail_test() {
...
   default_build_test_runner || fail_test "Control plane build tests failed"
-
-  header "Running data plane build tests"
-  data_plane_build_tests || fail_test "Data plane build tests failed"
 }
 
 function unit_tests() {
   header "Running control plane unit tests"
-  default_unit_test_runner || fail_test "Control plane unit tests failed"
+  #default_unit_test_runner || fail_test "Control plane unit tests failed"
```
Comment (pierDipi) [inline test/presubmit-tests.sh:42]: You commented go tests, this is the reason.

**Comment by pierDipi** [comment]
> Also, the tests report doesn't show the log lines related to each test (not a big problem)

I think this is more related to the report we're generating

File: hack/data-plane.sh [hunk 2]
```diff
@@ -145,30 +145,23 @@ function k8s() {
...
-function java_ut() {
-  docker build \
-    --file ${DATA_PLANE_DIR}/docker/test/Dockerfile \
-    --build-arg JAVA_IMAGE=${JAVA_IMAGE} \
-    --tag tests ${DATA_PLANE_DIR}
+  find . -type f -regextype posix-extended -regex ".*/TEST-.*.xml$" | xargs -I '{}' cp {} ${ARTIFACTS}/
+  pushd ${ARTIFACTS} || fail_test
+  for f in * ; do
+    mv -- "$f" "junit_$f" ;
+  done
```
[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply hack/data-plane.sh:157]: I think the last commit should do that... I think :smile:

**Comment by slinkydeveloper** [comment]
> I think this is more related to the report we're generating

We're not tweaking anything there, it's a standard junit5 report...

**Comment by pierDipi** [comment]
Yes, and are logs included in the standard report?

File: hack/data-plane.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (pierDipi) [reply hack/data-plane.sh:157]: We'll see. :)

**Review by pierDipi — APPROVED** [review]

**Comment by pierDipi** [comment]
/test pull-knative-sandbox-eventing-kafka-broker-unit-tests
--

**Comment by pierDipi** [comment]
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.2.1:shade (default) on project benchmarks: Execution default of goal org.apache.maven.plugins:maven-shade-plugin:3.2.1:shade failed: Plugin org.apache.maven.plugins:maven-shade-plugin:3.2.1 or one of its dependencies could not be resolved: Could not transfer artifact org.ow2.asm:asm:jar:7.0 from/to central (https://repo.maven.apache.org/maven2): transfer failed for https://repo.maven.apache.org/maven2/org/ow2/asm/asm/7.0/asm-7.0.jar: Connection reset

/test pull-knative-sandbox-eventing-kafka-broker-unit-tests

File: hack/data-plane.sh [hunk 3]
```diff
@@ -145,28 +145,20 @@ function k8s() {
 }
 
 function data_plane_unit_tests() {
-
-  java_ut || java_ut || fail_test "Data plane unit tests failed"
+  pushd ${DATA_PLANE_DIR} || fail_test
+  mvn clean verify --no-transfer-progress
```
Comment (pierDipi) [inline hack/data-plane.sh:149]: I think we need to retry here as we did before to mitigate connection reset errors

**Pull request merged**